src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -645,6 +645,64 @@ class FunctionBinaryArithmetic : public IFunction
         return FunctionFactory::instance().get(function_name, context);
     }
 
+    static FunctionOverloadResolverPtr
+    getFunctionForTupleArithmetic(const DataTypePtr & type0, const DataTypePtr & type1, ContextPtr context)
+    {
+        if (!isTuple(type0) || !isTuple(type1))
+            return {};
+
+        /// Special case when the function is plus, minus or multiply, both arguments are tuples.
+        /// We construct another function (example: tuplePlus) and call it.
+
+        if constexpr (!is_plus && !is_minus && !is_multiply)
+            return {};
+
+        std::string function_name;
+        if (is_plus)
+        {
+            function_name = "tuplePlus";
+        }
+        else if (is_minus)
+        {
+            function_name = "tupleMinus";
+        }
+        else
+        {
+            function_name = "dotProduct";
+        }
+
+        return FunctionFactory::instance().get(function_name, context);
+    }
+
+    static FunctionOverloadResolverPtr
+    getFunctionForTupleAndNumberArithmetic(const DataTypePtr & type0, const DataTypePtr & type1, ContextPtr context)
+    {
+        if (!(isTuple(type0) && isNumber(type1)) && !(isTuple(type1) && isNumber(type0)))
+            return {};
+
+        /// Special case when the function is multiply or divide, one of arguments is Tuple and another is Number.
+        /// We construct another function (example: tupleMultiplyByNumber) and call it.
+
+        if constexpr (!is_multiply && !is_division)
+            return {};
+
+        if (isNumber(type0) && is_division)
+            throw Exception("Wrong order of arguments for function " + String(name) + ": argument of numeric type cannot be first.",
```

Comment:
```suggestion
            throw Exception("Wrong order of arguments for function " + String(name) + ": argument of numeric type cannot be first",
```

Reply:
There was a dot in the similar exception in `getFunctionForIntervalArithmetic` (function in this file which was written not by me). I can remove dots in both exceptions.

Reply:
Yes, let's remove it because sometime exception message is wrapped with other one and we will got something like `argument of numeric type cannot be first.: <some additional info>`